pkg/prowgen/prowgen.go

```diff
@@ -266,6 +257,19 @@ func generatePresubmitForTest(jobBaseBuilder *prowJobBaseBuilder, name string, i
 
 	shortName := info.TestName(name)
 	base := jobBaseBuilder.Rehearsable(!opts.disableRehearsal).Build(jc.PresubmitPrefix)
+
+	// Set slack reporter config using full job name for proper excluded_job_patterns matching
+	fullJobName := info.JobName(jc.PresubmitPrefix, name)
+	if slackReporter := info.Config.GetSlackReporterConfigForJobName(fullJobName, name, info.Metadata.Variant); slackReporter != nil {
```

Comment:
This code has been duplicated three times. You might want to consider extracting it into its own function:
```golang
func addSlackReporterConfig(base *prowconfig.JobBase, jobName, testName string, info *ProwgenInfo) {
	if slackReporter := info.Config.GetSlackReporterConfigForJobName(jobName, testName, info.Metadata.Variant); slackReporter != nil {
		if base.ReporterConfig == nil {
			base.ReporterConfig = &prowv1.ReporterConfig{}
		}
		base.ReporterConfig.Slack = &prowv1.SlackReporterConfig{
			Channel:           slackReporter.Channel,
			JobStatesToReport: slackReporter.JobStatesToReport,
			ReportTemplate:    slackReporter.ReportTemplate,
		}
	}
}
```